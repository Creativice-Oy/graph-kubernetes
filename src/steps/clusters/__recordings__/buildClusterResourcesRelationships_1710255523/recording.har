{
  "log": {
    "_recordingName": "buildClusterResourcesRelationships",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "5.1.1"
    },
    "entries": [
      {
        "_id": "9d73307742273594a28ec2e4efce503e",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "name": "accept",
              "value": "application/json"
            },
            {
              "name": "host",
              "value": "192.168.49.2:8443"
            }
          ],
          "headersSize": 148,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "50"
            }
          ],
          "url": "https://192.168.49.2:8443/api/v1/namespaces?limit=50"
        },
        "response": {
          "bodySize": 1867,
          "content": {
            "mimeType": "application/json",
            "size": 1867,
            "text": "{\"kind\":\"NamespaceList\",\"apiVersion\":\"v1\",\"metadata\":{\"selfLink\":\"/api/v1/namespaces\",\"resourceVersion\":\"9161772\"},\"items\":[{\"metadata\":{\"name\":\"default\",\"selfLink\":\"/api/v1/namespaces/default\",\"uid\":\"9c533ce3-b6bd-43e6-947a-f9f73dc297b5\",\"resourceVersion\":\"155\",\"creationTimestamp\":\"2021-05-13T19:34:38Z\",\"managedFields\":[{\"manager\":\"kube-apiserver\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"2021-05-13T19:34:38Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:status\":{\"f:phase\":{}}}}]},\"spec\":{\"finalizers\":[\"kubernetes\"]},\"status\":{\"phase\":\"Active\"}},{\"metadata\":{\"name\":\"kube-node-lease\",\"selfLink\":\"/api/v1/namespaces/kube-node-lease\",\"uid\":\"11a8778d-4ef0-437b-bb1d-db3cf0a26e82\",\"resourceVersion\":\"33\",\"creationTimestamp\":\"2021-05-13T19:34:36Z\",\"managedFields\":[{\"manager\":\"kube-apiserver\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"2021-05-13T19:34:36Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:status\":{\"f:phase\":{}}}}]},\"spec\":{\"finalizers\":[\"kubernetes\"]},\"status\":{\"phase\":\"Active\"}},{\"metadata\":{\"name\":\"kube-public\",\"selfLink\":\"/api/v1/namespaces/kube-public\",\"uid\":\"ba4e6047-864a-46f9-9f29-3ffe004d7f13\",\"resourceVersion\":\"25\",\"creationTimestamp\":\"2021-05-13T19:34:36Z\",\"managedFields\":[{\"manager\":\"kube-apiserver\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"2021-05-13T19:34:36Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:status\":{\"f:phase\":{}}}}]},\"spec\":{\"finalizers\":[\"kubernetes\"]},\"status\":{\"phase\":\"Active\"}},{\"metadata\":{\"name\":\"kube-system\",\"selfLink\":\"/api/v1/namespaces/kube-system\",\"uid\":\"de2a4eee-505b-49d7-bc08-09026c88dae5\",\"resourceVersion\":\"16\",\"creationTimestamp\":\"2021-05-13T19:34:36Z\",\"managedFields\":[{\"manager\":\"kube-apiserver\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"2021-05-13T19:34:36Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:status\":{\"f:phase\":{}}}}]},\"spec\":{\"finalizers\":[\"kubernetes\"]},\"status\":{\"phase\":\"Active\"}}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "cache-control",
              "value": "no-cache, private"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "date",
              "value": "Wed, 03 Nov 2021 15:16:26 GMT"
            },
            {
              "name": "content-length",
              "value": "1867"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 146,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-11-03T15:16:26.942Z",
        "time": 25,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 25
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
