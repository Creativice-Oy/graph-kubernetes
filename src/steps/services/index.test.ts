import { fetchServices } from '.';
import { fetchNamespaces } from '../namespaces';
import { createDataCollectionTest } from '../../../test/recording';
import { integrationConfig } from '../../../test/config';
import { Entities, Relationships } from '../constants';
import { RelationshipClass } from '@jupiterone/data-model';

describe('#fetchServices', () => {
  test('should collect data', async () => {
    await createDataCollectionTest({
      recordingName: 'fetchServices',
      recordingDirectory: __dirname,
      integrationConfig,
      stepFunctions: [fetchNamespaces, fetchServices],
      entitySchemaMatchers: [
        {
          _type: Entities.SERVICE._type,
          matcher: {
            _class: ['Service'],
            schema: {
              additionalProperties: false,
              properties: {
                _type: { const: 'kube_service' },
                _rawData: {
                  type: 'array',
                  items: { type: 'object' },
                },
                name: { type: 'string' },
                displayName: { type: 'string' },
                generation: { type: 'number' },
                deletionGracePeriodSeconds: { type: 'string' },
                resourceVersion: { type: 'string' },
                category: {
                  type: 'array',
                  items: { type: 'string' },
                },
                createdOn: { type: 'number' },

                allocateLoadBalancerNodePorts: { type: 'boolean' },
                clusterIP: { type: 'string' },
                clusterIPs: {
                  type: 'array',
                  items: { type: 'string' },
                },
                externalIPs: {
                  type: 'array',
                  items: { type: 'string' },
                },
                externalName: { type: 'string' },
                externalTrafficPolicy: { type: 'string' },
                healthCheckNodePort: { type: 'number' },
                ipFamilies: {
                  type: 'array',
                  items: { type: 'string' },
                },
                ipFamilyPolicy: { type: 'string' },
                loadBalancerIP: { type: 'string' },
                loadBalancerSourceRanges: {
                  type: 'array',
                  items: { type: 'string' },
                },
                publishNotReadyAddresses: { type: 'boolean' },
                sessionAffinity: { type: 'string' },
                type: { type: 'string' },
              },
            },
          },
        },
      ],
      relationshipSchemaMatchers: [
        {
          _type: Relationships.NAMESPACE_HAS_SERVICE._type,
          matcher: {
            schema: {
              properties: {
                _class: { const: RelationshipClass.HAS },
                _type: { const: 'kube_namespace_has_service' },
              },
            },
          },
        },
      ],
    });
  });
});
