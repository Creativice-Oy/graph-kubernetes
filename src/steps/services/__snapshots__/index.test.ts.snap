// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchServices should collect data: jobState 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Group",
      ],
      "_key": "d5976ece-dc03-46a1-921f-cb6043e5f56e",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": V1Namespace {
            "apiVersion": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": undefined,
              "clusterName": undefined,
              "creationTimestamp": 2021-04-27T20:57:19.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": undefined,
              "labels": undefined,
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:status": Object {
                      "f:phase": Object {},
                    },
                  },
                  "manager": "kube-apiserver",
                  "operation": "Update",
                  "time": 2021-04-27T20:57:19.000Z,
                },
              ],
              "name": "default",
              "namespace": undefined,
              "ownerReferences": undefined,
              "resourceVersion": "200",
              "selfLink": undefined,
              "uid": "d5976ece-dc03-46a1-921f-cb6043e5f56e",
            },
            "spec": V1NamespaceSpec {
              "finalizers": Array [
                "kubernetes",
              ],
            },
            "status": V1NamespaceStatus {
              "conditions": undefined,
              "phase": "Active",
            },
          },
        },
      ],
      "_type": "kube_namespace",
      "active": undefined,
      "createdOn": 1619557039000,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "default",
      "finalizers": Array [
        "kubernetes",
      ],
      "generation": undefined,
      "name": "default",
      "resourceVersion": "200",
      "status.phase": "Active",
    },
    Object {
      "_class": Array [
        "Group",
      ],
      "_key": "55ea52dd-382c-4b8c-8c24-bff59db4b817",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": V1Namespace {
            "apiVersion": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": undefined,
              "clusterName": undefined,
              "creationTimestamp": 2021-04-27T20:57:17.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": undefined,
              "labels": undefined,
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:status": Object {
                      "f:phase": Object {},
                    },
                  },
                  "manager": "kube-apiserver",
                  "operation": "Update",
                  "time": 2021-04-27T20:57:17.000Z,
                },
              ],
              "name": "kube-node-lease",
              "namespace": undefined,
              "ownerReferences": undefined,
              "resourceVersion": "66",
              "selfLink": undefined,
              "uid": "55ea52dd-382c-4b8c-8c24-bff59db4b817",
            },
            "spec": V1NamespaceSpec {
              "finalizers": Array [
                "kubernetes",
              ],
            },
            "status": V1NamespaceStatus {
              "conditions": undefined,
              "phase": "Active",
            },
          },
        },
      ],
      "_type": "kube_namespace",
      "active": undefined,
      "createdOn": 1619557037000,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "kube-node-lease",
      "finalizers": Array [
        "kubernetes",
      ],
      "generation": undefined,
      "name": "kube-node-lease",
      "resourceVersion": "66",
      "status.phase": "Active",
    },
    Object {
      "_class": Array [
        "Group",
      ],
      "_key": "35ca0e47-73b0-4e3d-9e2a-d8f3d6506d88",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": V1Namespace {
            "apiVersion": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": undefined,
              "clusterName": undefined,
              "creationTimestamp": 2021-04-27T20:57:17.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": undefined,
              "labels": undefined,
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:status": Object {
                      "f:phase": Object {},
                    },
                  },
                  "manager": "kube-apiserver",
                  "operation": "Update",
                  "time": 2021-04-27T20:57:17.000Z,
                },
              ],
              "name": "kube-public",
              "namespace": undefined,
              "ownerReferences": undefined,
              "resourceVersion": "58",
              "selfLink": undefined,
              "uid": "35ca0e47-73b0-4e3d-9e2a-d8f3d6506d88",
            },
            "spec": V1NamespaceSpec {
              "finalizers": Array [
                "kubernetes",
              ],
            },
            "status": V1NamespaceStatus {
              "conditions": undefined,
              "phase": "Active",
            },
          },
        },
      ],
      "_type": "kube_namespace",
      "active": undefined,
      "createdOn": 1619557037000,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "kube-public",
      "finalizers": Array [
        "kubernetes",
      ],
      "generation": undefined,
      "name": "kube-public",
      "resourceVersion": "58",
      "status.phase": "Active",
    },
    Object {
      "_class": Array [
        "Group",
      ],
      "_key": "239ee6a3-23d6-4ca6-ae32-3b3fdda6a604",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": V1Namespace {
            "apiVersion": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": undefined,
              "clusterName": undefined,
              "creationTimestamp": 2021-04-27T20:57:17.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": undefined,
              "labels": undefined,
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:status": Object {
                      "f:phase": Object {},
                    },
                  },
                  "manager": "kube-apiserver",
                  "operation": "Update",
                  "time": 2021-04-27T20:57:17.000Z,
                },
              ],
              "name": "kube-system",
              "namespace": undefined,
              "ownerReferences": undefined,
              "resourceVersion": "34",
              "selfLink": undefined,
              "uid": "239ee6a3-23d6-4ca6-ae32-3b3fdda6a604",
            },
            "spec": V1NamespaceSpec {
              "finalizers": Array [
                "kubernetes",
              ],
            },
            "status": V1NamespaceStatus {
              "conditions": undefined,
              "phase": "Active",
            },
          },
        },
      ],
      "_type": "kube_namespace",
      "active": undefined,
      "createdOn": 1619557037000,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "kube-system",
      "finalizers": Array [
        "kubernetes",
      ],
      "generation": undefined,
      "name": "kube-system",
      "resourceVersion": "34",
      "status.phase": "Active",
    },
    Object {
      "_class": Array [
        "Service",
      ],
      "_key": "c296a0d8-9839-4908-bd52-971ddaaebcaf",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": V1Service {
            "apiVersion": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": undefined,
              "clusterName": undefined,
              "creationTimestamp": 2021-04-27T20:57:19.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": undefined,
              "labels": Object {
                "component": "apiserver",
                "provider": "kubernetes",
              },
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:metadata": Object {
                      "f:labels": Object {
                        ".": Object {},
                        "f:component": Object {},
                        "f:provider": Object {},
                      },
                    },
                    "f:spec": Object {
                      "f:clusterIP": Object {},
                      "f:ipFamilyPolicy": Object {},
                      "f:ports": Object {
                        ".": Object {},
                        "k:{\\"port\\":443,\\"protocol\\":\\"TCP\\"}": Object {
                          ".": Object {},
                          "f:name": Object {},
                          "f:port": Object {},
                          "f:protocol": Object {},
                          "f:targetPort": Object {},
                        },
                      },
                      "f:sessionAffinity": Object {},
                      "f:type": Object {},
                    },
                  },
                  "manager": "kube-apiserver",
                  "operation": "Update",
                  "time": 2021-04-27T20:57:19.000Z,
                },
              ],
              "name": "kubernetes",
              "namespace": "default",
              "ownerReferences": undefined,
              "resourceVersion": "202",
              "selfLink": undefined,
              "uid": "c296a0d8-9839-4908-bd52-971ddaaebcaf",
            },
            "spec": V1ServiceSpec {
              "allocateLoadBalancerNodePorts": undefined,
              "clusterIP": "10.96.0.1",
              "clusterIPs": Array [
                "10.96.0.1",
              ],
              "externalIPs": undefined,
              "externalName": undefined,
              "externalTrafficPolicy": undefined,
              "healthCheckNodePort": undefined,
              "ipFamilies": undefined,
              "ipFamilyPolicy": undefined,
              "loadBalancerIP": undefined,
              "loadBalancerSourceRanges": undefined,
              "ports": Array [
                V1ServicePort {
                  "appProtocol": undefined,
                  "name": "https",
                  "nodePort": undefined,
                  "port": 443,
                  "protocol": "TCP",
                  "targetPort": 8443,
                },
              ],
              "publishNotReadyAddresses": undefined,
              "selector": undefined,
              "sessionAffinity": "None",
              "sessionAffinityConfig": undefined,
              "topologyKeys": undefined,
              "type": "ClusterIP",
            },
            "status": V1ServiceStatus {
              "conditions": undefined,
              "loadBalancer": V1LoadBalancerStatus {
                "ingress": undefined,
              },
            },
          },
        },
      ],
      "_type": "kube_service",
      "active": undefined,
      "allocateLoadBalancerNodePorts": undefined,
      "category": Array [
        "infrastructure",
      ],
      "clusterIP": "10.96.0.1",
      "clusterIPs": Array [
        "10.96.0.1",
      ],
      "createdOn": 1619557039000,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "kubernetes",
      "externalIPs": undefined,
      "externalName": undefined,
      "externalTrafficPolicy": undefined,
      "generation": undefined,
      "healthCheckNodePort": undefined,
      "ipFamilies": undefined,
      "ipFamilyPolicy": undefined,
      "loadBalancerIP": undefined,
      "loadBalancerSourceRanges": undefined,
      "name": "kubernetes",
      "publishNotReadyAddresses": undefined,
      "resourceVersion": "202",
      "sessionAffinity": "None",
      "type": "ClusterIP",
    },
    Object {
      "_class": Array [
        "Service",
      ],
      "_key": "d44507f3-7f7b-411f-a49b-42dc1afbcae3",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": V1Service {
            "apiVersion": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": Object {
                "prometheus.io/port": "9153",
                "prometheus.io/scrape": "true",
              },
              "clusterName": undefined,
              "creationTimestamp": 2021-04-27T20:57:20.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": undefined,
              "labels": Object {
                "k8s-app": "kube-dns",
                "kubernetes.io/cluster-service": "true",
                "kubernetes.io/name": "KubeDNS",
              },
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:metadata": Object {
                      "f:annotations": Object {
                        ".": Object {},
                        "f:prometheus.io/port": Object {},
                        "f:prometheus.io/scrape": Object {},
                      },
                      "f:labels": Object {
                        ".": Object {},
                        "f:k8s-app": Object {},
                        "f:kubernetes.io/cluster-service": Object {},
                        "f:kubernetes.io/name": Object {},
                      },
                    },
                    "f:spec": Object {
                      "f:clusterIP": Object {},
                      "f:ports": Object {
                        ".": Object {},
                        "k:{\\"port\\":53,\\"protocol\\":\\"TCP\\"}": Object {
                          ".": Object {},
                          "f:name": Object {},
                          "f:port": Object {},
                          "f:protocol": Object {},
                          "f:targetPort": Object {},
                        },
                        "k:{\\"port\\":53,\\"protocol\\":\\"UDP\\"}": Object {
                          ".": Object {},
                          "f:name": Object {},
                          "f:port": Object {},
                          "f:protocol": Object {},
                          "f:targetPort": Object {},
                        },
                        "k:{\\"port\\":9153,\\"protocol\\":\\"TCP\\"}": Object {
                          ".": Object {},
                          "f:name": Object {},
                          "f:port": Object {},
                          "f:protocol": Object {},
                          "f:targetPort": Object {},
                        },
                      },
                      "f:selector": Object {
                        ".": Object {},
                        "f:k8s-app": Object {},
                      },
                      "f:sessionAffinity": Object {},
                      "f:type": Object {},
                    },
                  },
                  "manager": "kubeadm",
                  "operation": "Update",
                  "time": 2021-04-27T20:57:20.000Z,
                },
              ],
              "name": "kube-dns",
              "namespace": "kube-system",
              "ownerReferences": undefined,
              "resourceVersion": "241",
              "selfLink": undefined,
              "uid": "d44507f3-7f7b-411f-a49b-42dc1afbcae3",
            },
            "spec": V1ServiceSpec {
              "allocateLoadBalancerNodePorts": undefined,
              "clusterIP": "10.96.0.10",
              "clusterIPs": Array [
                "10.96.0.10",
              ],
              "externalIPs": undefined,
              "externalName": undefined,
              "externalTrafficPolicy": undefined,
              "healthCheckNodePort": undefined,
              "ipFamilies": undefined,
              "ipFamilyPolicy": undefined,
              "loadBalancerIP": undefined,
              "loadBalancerSourceRanges": undefined,
              "ports": Array [
                V1ServicePort {
                  "appProtocol": undefined,
                  "name": "dns",
                  "nodePort": undefined,
                  "port": 53,
                  "protocol": "UDP",
                  "targetPort": 53,
                },
                V1ServicePort {
                  "appProtocol": undefined,
                  "name": "dns-tcp",
                  "nodePort": undefined,
                  "port": 53,
                  "protocol": "TCP",
                  "targetPort": 53,
                },
                V1ServicePort {
                  "appProtocol": undefined,
                  "name": "metrics",
                  "nodePort": undefined,
                  "port": 9153,
                  "protocol": "TCP",
                  "targetPort": 9153,
                },
              ],
              "publishNotReadyAddresses": undefined,
              "selector": Object {
                "k8s-app": "kube-dns",
              },
              "sessionAffinity": "None",
              "sessionAffinityConfig": undefined,
              "topologyKeys": undefined,
              "type": "ClusterIP",
            },
            "status": V1ServiceStatus {
              "conditions": undefined,
              "loadBalancer": V1LoadBalancerStatus {
                "ingress": undefined,
              },
            },
          },
        },
      ],
      "_type": "kube_service",
      "active": undefined,
      "allocateLoadBalancerNodePorts": undefined,
      "category": Array [
        "infrastructure",
      ],
      "clusterIP": "10.96.0.10",
      "clusterIPs": Array [
        "10.96.0.10",
      ],
      "createdOn": 1619557040000,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "kube-dns",
      "externalIPs": undefined,
      "externalName": undefined,
      "externalTrafficPolicy": undefined,
      "generation": undefined,
      "healthCheckNodePort": undefined,
      "ipFamilies": undefined,
      "ipFamilyPolicy": undefined,
      "loadBalancerIP": undefined,
      "loadBalancerSourceRanges": undefined,
      "name": "kube-dns",
      "publishNotReadyAddresses": undefined,
      "resourceVersion": "241",
      "sessionAffinity": "None",
      "type": "ClusterIP",
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "HAS",
      "_fromEntityKey": "d5976ece-dc03-46a1-921f-cb6043e5f56e",
      "_key": "d5976ece-dc03-46a1-921f-cb6043e5f56e|has|c296a0d8-9839-4908-bd52-971ddaaebcaf",
      "_toEntityKey": "c296a0d8-9839-4908-bd52-971ddaaebcaf",
      "_type": "kube_namespace_has_service",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "239ee6a3-23d6-4ca6-ae32-3b3fdda6a604",
      "_key": "239ee6a3-23d6-4ca6-ae32-3b3fdda6a604|has|d44507f3-7f7b-411f-a49b-42dc1afbcae3",
      "_toEntityKey": "d44507f3-7f7b-411f-a49b-42dc1afbcae3",
      "_type": "kube_namespace_has_service",
      "displayName": "HAS",
    },
  ],
  "encounteredTypes": Array [
    "kube_namespace",
    "kube_service",
    "kube_namespace_has_service",
  ],
  "numCollectedEntities": 6,
  "numCollectedRelationships": 2,
}
`;
